File: 프로그래머스/unrated/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    N = len(sequence)
    end = 0
    interval_sum = 0
    result = []
    for start in range(N):
        while interval_sum < k and end < N:
            interval_sum += sequence[end]
            end += 1
        if interval_sum == k:
            if not result:
                result = [start, end-1]
            elif result[1]-result[0] > end-1 - start:
                result = [start, end-1]
        interval_sum -= sequence[start]

    return result

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int n = sequence.length;
        int sum = 0;
        int end = 0;

        int left = -1;
        int right = -1;

        for (int start = 0; start < n; start++) {
            while (sum < k && end < n) {
                sum += sequence[end];
                end += 1;
            }

            if (sum == k) {
                if (left == -1) {
                    left = start;
                    right = end - 1;
                } else {
                    if (right - left > end - 1 - start) {
                        left = start;
                        right = end - 1;
                    }
                }
            }
            
            sum -= sequence[start];
        }

        return new int[]{left, right};
    }
}